CONTRIBUTING.md

```diff
@@ -0,0 +1,77 @@
+# Contribution Guidelines
+
+Thank you for your interest in contributing to our project :)<br>
+We like issue reports, suggestions and pull request from everyone.
+
+If you are planning to make a major contribution for which there is no existing issue, we would appreciate that you open an issue first.
+
+The following suggestions increase the chance that your pull request can be accepted:
+
+-   Write clean code
+-   Write tests for your new code
+-   Check that all tests are still passed
+-   Adhere to the naming conventions for branch names and commit messages mentioned below
+-   Avoid unnecessary breaking changes
+
+You and your code contribution has to adherence to the licence of this project.
+
+## Naming Conventions
+
+### Branch Names
+
+Branch names should follow the following structure: `<type>/<issue-id>/<name>`
+
+-   `<type>` is the type of the change (see table below).
+-   `<number>` is the number of the corresponding issue on GitHub the branch should resolve.
+-   `<name>` is a describing branch name, which should be in **lowercase** separated by **dashes**.
+
+### Allowed Change Types
+
+| Change Type | Description                                    |
+| ----------- | ---------------------------------------------- |
+| `feat`      | Adds a new or expands an existing feature      |
+| `fix`       | Fixes a bug                                    |
+| `docs`      | Updates documentation only                     |
+| `style`     | Code style improvements, no logic changes      |
+| `refactor`  | Refactorings (neither "feat" nor "fix")        |
+| `perf`      | Performance optimizations                      |
+| `test`      | Adds or improves test cases                    |
+| `build`     | Build system or external dependency changes    |
+| `ci`        | Changes to the CI configuration or scripts     |
+| `chore`     | Other changes, do not modify src or test files |
+| `revert`    | Reverts a previous commit                      |
+
+Examples:
+
+-   `feat/123/settings-option-xyz`
+-   `fix/124/unexpected-settings-error`
+
+### Commit Messages
```

Comment:
I would say that normal commits don#t have to follow this conventions, since they contain less significant changes. But the merge commit should follow this, because it submits the changes into the main.

Reply:
See also https://www.conventionalcommits.org/en/v1.0.0/#do-all-my-contributors-need-to-use-the-conventional-commits-specification